zipline/pipeline/engine.py

```diff
@@ -315,10 +329,13 @@ def compute_chunk(self, graph, dates, assets, initial_workspace):
             mask, mask_dates = self._mask_and_dates_for_term(
                 term, workspace, graph, dates
             )
+
             if term.atomic:
-                # FUTURE OPTIMIZATION: Scan the resolution order for terms in
-                # the same dataset and load them here as well.
-                to_load = [term]
+                to_load = sorted(
+                    atomic_groups[atomic_group_key(term)],
+                    key=lambda t: t.dataset
```

Comment:
I actually have a test for it: https://github.com/quantopian/zipline/pull/756/files#diff-9b17b1ed8b53c49c54dccce45979890fR463